### PR TITLE
Userinfo API

### PR DESCRIPTION
**Pull request by pubudu538**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/auth/org.wso2.carbon.auth.user.info/pom.xml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ /*
+  ~ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mushthaq33) [inline components/auth/org.wso2.carbon.auth.user.info/pom.xml]: Old License header. Shall we change it to the latest one?

File: components/auth/org.wso2.carbon.auth.user.info/src/main/java/org/wso2/carbon/auth/user/info/internal/ConfigurationActivator.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package org.wso2.carbon.auth.user.info.internal;
+/*
```
Comment (mushthaq33) [inline components/auth/org.wso2.carbon.auth.user.info/src/main/java/org/wso2/carbon/auth/user/info/internal/ConfigurationActivator.java]: The license header needs to be on top of the package right? Shall we change it?

File: components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/exception/ExceptionCodes.java [hunk 3]
```diff
@@ -36,7 +36,10 @@
 
     //oauth2 grants related exception codes
     OAUTH2_GRANT_PROCESS_EXCEPTION(900500, "Internal OAuth2 grant processing error", 500,
-            "Error while processing OAuth2 grant request");
+            "Error while processing OAuth2 grant request"),
+    INVALID_REQUEST(900600, "Invalid request", 400, "Invalid request"),
+    UNSUPPORTED_SCOPE(900601, "Unsupported scope", 400, "Unsupported scope"),
```
Comment (Rajith90) [inline components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/exception/ExceptionCodes.java]: Word Unsupported can be misleading, AFAIU there are no unsupported scopes, there can be invalid scopes.

Comment (pubudu538) [reply components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/exception/ExceptionCodes.java]: Userinfo only supports openid scope. When you provide a token for userinfo, that token's scope should be openid. You can generate tokens with different scopes. But userinfo API will only work for tokens with openid scope. That's why there is an unsupported scope.

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**